test/suite/test_prepare_hs04.py

```diff
@@ -0,0 +1,217 @@
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from helper import copy_wiredtiger_home
+import wiredtiger, wttest
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+from wiredtiger import stat
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_prepare_hs04.py
+# Read prepared updates from on-disk with ignore_prepare.
+# Committing or aborting a prepared update when there exists a tombstone for that key already.
+#
+class test_prepare_hs04(wttest.WiredTigerTestCase):
+    # Force a small cache.
+    conn_config = 'cache_size=50MB,statistics=(fast)'
+
+    # Create a small table.
+    uri = "table:test_prepare_hs04"
+
+    nsessions = 3
+    nkeys = 40
+    nrows = 100
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def search_key_timestamp_and_ignore(self, ds, timestamp, ignore_prepare_str, after_crash=None):
+        cursor = self.session.open_cursor(self.uri)
+
+        txn_config = 'read_timestamp=' + timestamp_str(timestamp) + ',ignore_prepare=' + ignore_prepare_str
+        commit_key = "C"
+        self.session.begin_transaction(txn_config)
+        for i in range(1, self.nsessions * self.nkeys):
+            key = commit_key + ds.key(self.nrows + i)
+            cursor.set_key(key)
+            if timestamp <= 10:
+                self.assertEqual(cursor.search(), 0)
+            elif timestamp == 20 and ignore_prepare_str == "true":
+                if after_crash == True:
+                    # Cursor search should not find the key with ignore_prepare=true and after_crash
+                    # Because the prepared_updates are committed with timestamp 30
+                    self.assertEqual(cursor.search(), wiredtiger.WT_NOTFOUND)
+                else:
+                    # Cursor search should not find the key with ignore_prepare=true
+                    self.assertEqual(cursor.search(), wiredtiger.WT_NOTFOUND)
+            elif timestamp == 20 and ignore_prepare_str == "false":
+                if after_crash == True:
+                    # Cursor search should not find the key with ignore_prepare=false and after_crash
+                    # Because the prepared_updates are committed with timestamp 30
+                    self.assertEqual(cursor.search(), wiredtiger.WT_NOTFOUND)
+                else:
+                    preparemsg = '/conflict with a prepared update/'
+                    # Make sure we get the expected prepare conflict message.
+                    self.assertRaisesException(wiredtiger.WiredTigerError, lambda:cursor.search(), preparemsg)
+            elif timestamp == 30:
+                # Keys are visible at timestamp 30.
+                self.assertEqual(cursor.search(), 0)
```

Comment:
I think this function would be clearer (and reusable if copied to future tests), if it didn't have "knowledge" of timestamp or prepare_str.  Rather, it should be told what to expect.  Perhaps
```
def search_keys_timestamp_and_ignore(self, ds, expected_value, conflict=False)
```
If expected_value is `None`, then expect `WT_NOTFOUND`. Otherwise, `search` returns 0 and `expected_value` is a strings and should match what's returned by `cursor.get_value()`.

The advantage of this approach is that this function doesn't have to know anything about the particulars of the caller.  The caller knows whether it should succeed or fail, what the expected value would be, and whether a conflict message is expected.

Reply:
@ddanderson Yeah, I agree and have changed the function signature as `def search_keys_timestamp_and_ignore(self, ds, txn_config, expected_value, conflict=False)` . I am constructing txn_config outside of the function with different timestamps as passing as a parameter.